src/goto-symex/goto_symex_state.cpp

```diff
@@ -452,8 +452,7 @@ bool goto_symex_statet::l2_thread_read_encoding(
       source,
       symex_targett::assignment_typet::PHI);
 
-    // TODO: are we setting l2 indices of something that is already l2?
-    expr = set_indices<L2>(std::move(ssa_l2), ns).get();
+    expr = std::move(ssa_l2);
```

Comment:
See my earlier comment: did you add an assertion to confirm this is doing what you expected across our regression tests?